ravenframework/SupervisedLearning/SyntheticHistory.py

```diff
@@ -97,6 +98,65 @@ def __evaluateLocal__(self, featureVals):
     rlz = self.evaluateTSASequential()
     return rlz
 
+
+  def getGlobalRomSegmentSettings(self, trainingDict, divisions):
+    """
+      Allows the ROM to perform some analysis before segmenting.
+      Note this is called on the GLOBAL templateROM from the ROMcollection, NOT on the LOCAL subsegment ROMs!
+      @ In, trainingDict, dict, data for training, full and unsegmented
+      @ In, divisions, tuple, (division slice indices, unclustered spaces)
+      @ Out, settings, object, arbitrary information about ROM clustering settings
+      @ Out, trainingDict, dict, adjusted training data (possibly unchanged)
+    """
+    self.raiseADebug('Training Global...')
+    # extracting info from training Dict, convert all signals to single array
+    trainingDict = copy.deepcopy(trainingDict)
+    names, values  = list(trainingDict.keys()), list(trainingDict.values())
+    ## This is for handling the special case needed by skl *MultiTask* that
+    ## requires multiple targets.
+    targetValues = []
+    targetNames = []
+    for target in self.target:
+      if target in names:
+        targetValues.append(values[names.index(target)])
+        targetNames.append(target)
+      else:
+        self.raiseAnError(IOError,'The target '+target+' is not in the training set')
+    # stack targets
+    targetValues = np.stack(targetValues, axis=-1)
+    self.trainTSASequential(targetValues, trainGlobal=True)
+    settings = self.getGlobalTSARomSettings()
+    # update targets in trainingDict
+    for i,target in enumerate(targetNames):
+      trainingDict[target] = targetValues[:,:,i]
+    return settings, trainingDict
+
+  def finalizeGlobalRomSegmentEvaluation(self, settings, evaluation, weights, slicer):
+    """
+      Allows any global settings to be applied to the signal collected by the ROMCollection instance.
+      Note this is called on the GLOBAL templateROM from the ROMcollection, NOT on the LOCAL supspace segment ROMs!
+      @ In, evaluation, dict, {target: np.ndarray} evaluated full (global) signal from ROMCollection
+      TODO finish docs
+      @ Out, evaluation, dict, {target: np.ndarray} adjusted global evaluation
+    """
+    if len(self._tsaGlobalAlgorithms)>0:
+      rlz = self.evaluateTSASequential(evalGlobal=True, evaluation=evaluation, slicer=slicer)
+      for key,val in rlz.items():
+        evaluation[key] = val
+    return evaluation
+
+  def finalizeLocalRomSegmentEvaluation(self,  settings, evaluation, globalPicker, localPicker=None):
```

Comment:
Is this a required method to satisfy some abstract method definition? Seems odd to just return the input `evaluation`

Reply:
I had overloaded the method from `SupervisedLearning` because the inputs were different, but comparing with the `finalizeLocalRomSegmentEvaluation( )` method in the `SupervisedLearning.ARMA` it seems that the ARMA version just has an additional optional input that had not been updated in the parent version. 

I am removing this method from `SyntheticHistory.py`, updating the method in `SupervisedLearning.py` with the additional optional input it is missing so that it functions as the "abstract/do nothing" parent method (since it also just returns `evaluation`) for TSA